### PR TITLE
fix minor mfs truncate bug

### PR DESCRIPTION
**Pull request by whyrusleeping**
Thanks to @RichardLitt for saying "truncate doesnt work" and making go try to prove him wrong and find a bug in the process.

License: MIT
Signed-off-by: Jeromy <jeromyj@gmail.com>

### CODE REVIEW

**Comment by whyrusleeping** [comment]
@noffle CR please!

**Comment by RichardLitt** [comment]
:)

**Comment by noffle** [comment]
Some clarifying questions for context:

1. What was Richard's usage of truncate that caused the breakage?
2. What was the breakage?
3. What was the expected behaviour?

**Comment by RichardLitt** [comment]
1. I was trying to use `curl` to access the IPFS API, and truncate wasn't showing the responses I would have liked. https://github.com/ipfs/http-api-spec/blob/feature/files/apiary.apib#L3796
2. Lack of response from using truncate. See https://github.com/ipfs/go-ipfs/issues/2380
3. I didn't know, see https://github.com/ipfs/go-ipfs/issues/2380

File: test/sharness/t0250-files-api.sh [hunk 1]
```diff
@@ -358,6 +358,25 @@ test_files_api() {
 		verify_dir_contents /
 	'
 
+	# test truncating
+	test_expect_success "create a new file" '
+		echo "some content" | ipfs files write --create /cats
```
Comment (noffle) [inline test/sharness/t0250-files-api.sh]: Perhaps also verify its contents are as expected before doing the truncate below?

File: mfs/dir.go [hunk 2]
```diff
@@ -126,7 +126,7 @@ func (d *Directory) childNode(name string) (FSNode, error) {
 		ndir := NewDirectory(d.ctx, name, nd, d, d.dserv)
 		d.childDirs[name] = ndir
 		return ndir, nil
-	case ufspb.Data_File:
+	case ufspb.Data_File, ufspb.Data_Raw:
```
Comment (noffle) [inline mfs/dir.go]: Hey @whyrusleeping, doesn't this fix imply that the `truncate` option is changing the file type from `Data_File` to `Data_Raw`? Is this intended?

**Pull request merged**